accepted/00074-retracted-updates-support.md

```diff
@@ -0,0 +1,307 @@
+- has an installed package `hello-2.10-lp151.2.6.x86_64`
+- is registered in a repository containing a retracted patch associated with
+  package `hello-2.10-lp151.4.1.x86_64` (newer than the installed one)
+
+Operations and their outcomes:
+- listing (`zypper patches`): lists retracted patches with status `retracted`
+- searching for new versions (`zypper search -v hello`) gets us:
+  `vR | hello | package | 2.10-lp151.4.1` <--- R means retracted
+- upgrading a package (`zypper up hello`):
+  does not upgrade to a version contained in a retracted patch
+- installing specific version (`zypper in hello-2.10-lp151.4.1`):
+  **allows** installing a package contained in a retracted patch (this must be
+  taken into account, since this is the way we upgrade packages in SUMA)
+- installing retracted patch (`zypper in patch:<retracted_patch_id>`):
+  yields "patch not needed" and exit code `0`. Consequence: retracted patch
+  can't be applied to a system! Q: somebody double-check this.
+- if a system already has a patch installed, that becomes retracted, zypper has
+  no intention to fix this situation
+
+## Iteration 1 - Minimal Viable Product
+
+### Database
+The `rhnErrata` table must be updated with the `status` column of type
+`VARCHAR(32)`.
+
+Q: Alternatives: 1) `NUMERIC`, 2) normalize the `status` attribute and use
+another table (see `rhnErrataSeverity` for an example).
+
+### Reposync
+
+The status `retracted` is part of `update` elements in the `updatedata.xml` file
+of the repository. We need to adjust `spacewalk-repo-sync` (`errataImport.py`)
+so that it parses this attribute during repository synchronization and stores it
+in the database (`rhnErrata.status`).
+
+Q: How does SCC retract the patch? Does it modify the patch with the same id by
+   changing `status=stable` to `status=retracted` or does it somehow issue a new
+   patch?
+   A: i think it's the former (modify), but verify that!
+   https://jira.suse.com/browse/SLE-8770
+Q: If the previous question==true: We need to think of consequences of modifying
+   an existing patch. Normally reposync is not designed to modify data. We need
+   to make sure reposync spots this change and maybe we also need to update some
+   caches (system available updates are cached (`rhnServerNeededCache`?)).
+
+### Java backend
+
+We need to enhance the `Errata` class with the `status` attribute, for which we
+will implement an `enum`.
+
+### Repository generating
+Enhance the `UpdateInfoWriter` class, so that it includes the `status` attribute
+in the output XML. Currently the `status` output is hardcoded to `final`.
+
+
+### UI
+
+The following pages must be updated:
+
+- Patch detail: information about the status
+Q: should this be read-write? so that user can modify the `status` attribute
+using a toggle/drop-down (for non-vendor channels). If we allow this, we need to
+face the consequences of modifying a patch (maybe regenerating some data
+(`rhnServerNeededCache`)).
```

Comment:
IIUC patches are fully editable from the UI, so it would make sense that this bit is editable as well.